Currently getting this error:

```bash
nix-build --attr pkgs.sketchybar
this derivation will be built:
  /nix/store/4pas9210ydg0nw2mij5mclx90p73x412-sketchybar-2.16.0.drv
building '/nix/store/4pas9210ydg0nw2mij5mclx90p73x412-sketchybar-2.16.0.drv'...
unpacking sources
unpacking source archive /nix/store/rd5lvshk0rfzdg6qi46kzk34lp7bgbww-source
source root is source
patching sources
updateAutotoolsGnuConfigScriptsPhase
configuring
no configure script, doing nothing
building
build flags: SHELL=/nix/store/wbpbgy7a09bicknf1yl397mqyyfs0kv7-bash-5.2-p15/bin/bash arm64
mkdir bin
clang -c -o bin/alias.o src/alias.c -std=c99 -Wall -Ofast -ffast-math -fvisibility=hidden -fno-common -target arm64-apple-macos11
clang -c -o bin/background.o src/background.c -std=c99 -Wall -Ofast -ffast-math -fvisibility=hidden -fno-common -target arm64-apple-macos11
clang -c -o bin/bar_item.o src/bar_item.c -std=c99 -Wall -Ofast -ffast-math -fvisibility=hidden -fno-common -target arm64-apple-macos11
clang -c -o bin/custom_events.o src/custom_events.c -std=c99 -Wall -Ofast -ffast-math -fvisibility=hidden -fno-common -target arm64-apple-macos11
clang -c -o bin/event.o src/event.c -std=c99 -Wall -Ofast -ffast-math -fvisibility=hidden -fno-common -target arm64-apple-macos11
clang -c -o bin/graph.o src/graph.c -std=c99 -Wall -Ofast -ffast-math -fvisibility=hidden -fno-common -target arm64-apple-macos11
clang -c -o bin/image.o src/image.c -std=c99 -Wall -Ofast -ffast-math -fvisibility=hidden -fno-common -target arm64-apple-macos11
clang -c -o bin/mouse.o src/mouse.c -std=c99 -Wall -Ofast -ffast-math -fvisibility=hidden -fno-common -target arm64-apple-macos11
clang -c -o bin/shadow.o src/shadow.c -std=c99 -Wall -Ofast -ffast-math -fvisibility=hidden -fno-common -target arm64-apple-macos11
clang -c -o bin/font.o src/font.c -std=c99 -Wall -Ofast -ffast-math -fvisibility=hidden -fno-common -target arm64-apple-macos11
clang -c -o bin/text.o src/text.c -std=c99 -Wall -Ofast -ffast-math -fvisibility=hidden -fno-common -target arm64-apple-macos11
clang -c -o bin/message.o src/message.c -std=c99 -Wall -Ofast -ffast-math -fvisibility=hidden -fno-common -target arm64-apple-macos11
clang -c -o bin/bar.o src/bar.c -std=c99 -Wall -Ofast -ffast-math -fvisibility=hidden -fno-common -target arm64-apple-macos11
clang -c -o bin/color.o src/color.c -std=c99 -Wall -Ofast -ffast-math -fvisibility=hidden -fno-common -target arm64-apple-macos11
clang -c -o bin/window.o src/window.c -std=c99 -Wall -Ofast -ffast-math -fvisibility=hidden -fno-common -target arm64-apple-macos11
clang -c -o bin/bar_manager.o src/bar_manager.c -std=c99 -Wall -Ofast -ffast-math -fvisibility=hidden -fno-common -target arm64-apple-macos11
clang -c -o bin/display.o src/display.c -std=c99 -Wall -Ofast -ffast-math -fvisibility=hidden -fno-common -target arm64-apple-macos11
clang -c -o bin/group.o src/group.c -std=c99 -Wall -Ofast -ffast-math -fvisibility=hidden -fno-common -target arm64-apple-macos11
clang -c -o bin/mach.o src/mach.c -std=c99 -Wall -Ofast -ffast-math -fvisibility=hidden -fno-common -target arm64-apple-macos11
clang -c -o bin/popup.o src/popup.c -std=c99 -Wall -Ofast -ffast-math -fvisibility=hidden -fno-common -target arm64-apple-macos11
clang -c -o bin/animation.o src/animation.c -std=c99 -Wall -Ofast -ffast-math -fvisibility=hidden -fno-common -target arm64-apple-macos11
clang -c -o bin/workspace.om src/workspace.m -std=c99 -Wall -Ofast -ffast-math -fvisibility=hidden -fno-common -target arm64-apple-macos11
clang -c -o bin/volume.o src/volume.c -std=c99 -Wall -Ofast -ffast-math -fvisibility=hidden -fno-common -target arm64-apple-macos11
clang -c -o bin/slider.o src/slider.c -std=c99 -Wall -Ofast -ffast-math -fvisibility=hidden -fno-common -target arm64-apple-macos11
clang -c -o bin/power.o src/power.c -std=c99 -Wall -Ofast -ffast-math -fvisibility=hidden -fno-common -target arm64-apple-macos11
clang -c -o bin/wifi.om src/wifi.m -std=c99 -Wall -Ofast -ffast-math -fvisibility=hidden -fno-common -target arm64-apple-macos11
clang -c -o bin/media.om src/media.m -std=c99 -Wall -Ofast -ffast-math -fvisibility=hidden -fno-common -target arm64-apple-macos11
src/media.m:87:25: error: expected a type
- (void)playing_change:(NSNotification *)notification {
                        ^
src/media.m:95:23: error: expected a type
- (void)media_change:(NSNotification *)notification {
                      ^
src/media.m:28:8: error: use of undeclared identifier 'NSNotificationCenter'
      [NSNotificationCenter.defaultCenter addObserver:self
       ^
src/media.m:33:8: error: use of undeclared identifier 'NSNotificationCenter'
      [NSNotificationCenter.defaultCenter addObserver:self
       ^
src/media.m:38:8: error: use of undeclared identifier 'NSNotificationCenter'
      [NSNotificationCenter.defaultCenter addObserver:self
       ^
src/media.m:51:40: warning: instance method '-UTF8String' not found (return type defaults to 'id') [-Wobjc-method-access]
      const char* app_cstr = [self.app UTF8String];
                                       ^~~~~~~~~~
/nix/store/km68nqhsk0h20ajnlqys3cnlqshs7cgv-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:238:8: note: receiver is instance of class declared here
@class NSString;
       ^
src/media.m:51:19: warning: incompatible pointer types initializing 'const char *' with an expression of type 'id' [-Wincompatible-pointer-types]
      const char* app_cstr = [self.app UTF8String];
                  ^          ~~~~~~~~~~~~~~~~~~~~~
src/media.m:52:46: warning: instance method '-UTF8String' not found (return type defaults to 'id') [-Wobjc-method-access]
      const char* artist_cstr = [self.artist UTF8String];
                                             ^~~~~~~~~~
/nix/store/km68nqhsk0h20ajnlqys3cnlqshs7cgv-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:238:8: note: receiver is instance of class declared here
@class NSString;
       ^
src/media.m:52:19: warning: incompatible pointer types initializing 'const char *' with an expression of type 'id' [-Wincompatible-pointer-types]
      const char* artist_cstr = [self.artist UTF8String];
                  ^             ~~~~~~~~~~~~~~~~~~~~~~~~
src/media.m:53:44: warning: instance method '-UTF8String' not found (return type defaults to 'id') [-Wobjc-method-access]
      const char* title_cstr = [self.title UTF8String];
                                           ^~~~~~~~~~
/nix/store/km68nqhsk0h20ajnlqys3cnlqshs7cgv-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:238:8: note: receiver is instance of class declared here
@class NSString;
       ^
src/media.m:53:19: warning: incompatible pointer types initializing 'const char *' with an expression of type 'id' [-Wincompatible-pointer-types]
      const char* title_cstr = [self.title UTF8String];
                  ^            ~~~~~~~~~~~~~~~~~~~~~~~
src/media.m:54:44: warning: instance method '-UTF8String' not found (return type defaults to 'id') [-Wobjc-method-access]
      const char* album_cstr = [self.album UTF8String];
                                           ^~~~~~~~~~
/nix/store/km68nqhsk0h20ajnlqys3cnlqshs7cgv-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:238:8: note: receiver is instance of class declared here
@class NSString;
       ^
src/media.m:54:19: warning: incompatible pointer types initializing 'const char *' with an expression of type 'id' [-Wincompatible-pointer-types]
      const char* album_cstr = [self.album UTF8String];
                  ^            ~~~~~~~~~~~~~~~~~~~~~~~
src/media.m:99:32: error: property 'userInfo' not found on object of type 'const id'
      self.app = [notification.userInfo valueForKey:kMRMediaRemoteNowPlayingApplicationDisplayNameUserInfoKey];
                               ^
src/media.m:100:27: warning: instance method '-valueForKey:' not found (return type defaults to 'id') [-Wobjc-method-access]
      self.artist = [dict valueForKey:kMRMediaRemoteNowPlayingInfoArtist];
                          ^~~~~~~~~~~
/nix/store/km68nqhsk0h20ajnlqys3cnlqshs7cgv-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:244:8: note: receiver is instance of class declared here
@class NSDictionary;
       ^
src/media.m:101:26: warning: instance method '-valueForKey:' not found (return type defaults to 'id') [-Wobjc-method-access]
      self.title = [dict valueForKey:kMRMediaRemoteNowPlayingInfoTitle];
                         ^~~~~~~~~~~
/nix/store/km68nqhsk0h20ajnlqys3cnlqshs7cgv-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:244:8: note: receiver is instance of class declared here
@class NSDictionary;
       ^
src/media.m:102:26: warning: instance method '-valueForKey:' not found (return type defaults to 'id') [-Wobjc-method-access]
      self.album = [dict valueForKey:kMRMediaRemoteNowPlayingInfoAlbum];
                         ^~~~~~~~~~~
/nix/store/km68nqhsk0h20ajnlqys3cnlqshs7cgv-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:244:8: note: receiver is instance of class declared here
@class NSDictionary;
       ^
11 warnings and 6 errors generated.
make: *** [makefile:61: bin/media.om] Error 1
error: builder for '/nix/store/4pas9210ydg0nw2mij5mclx90p73x412-sketchybar-2.16.0.drv' failed with exit code 2;
       last 50 log lines:
       > src/media.m:52:46: warning: instance method '-UTF8String' not found (return type defaults to 'id') [-Wobjc-method-access]
       >       const char* artist_cstr = [self.artist UTF8String];
       >                                              ^~~~~~~~~~
       > /nix/store/km68nqhsk0h20ajnlqys3cnlqshs7cgv-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:238:8: note: receiver is instance of class declared here
       > @class NSString;
       >        ^
       > src/media.m:52:19: warning: incompatible pointer types initializing 'const char *' with an expression of type 'id' [-Wincompatible-pointer-types]
       >       const char* artist_cstr = [self.artist UTF8String];
       >                   ^             ~~~~~~~~~~~~~~~~~~~~~~~~
       > src/media.m:53:44: warning: instance method '-UTF8String' not found (return type defaults to 'id') [-Wobjc-method-access]
       >       const char* title_cstr = [self.title UTF8String];
       >                                            ^~~~~~~~~~
       > /nix/store/km68nqhsk0h20ajnlqys3cnlqshs7cgv-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:238:8: note: receiver is instance of class declared here
       > @class NSString;
       >        ^
       > src/media.m:53:19: warning: incompatible pointer types initializing 'const char *' with an expression of type 'id' [-Wincompatible-pointer-types]
       >       const char* title_cstr = [self.title UTF8String];
       >                   ^            ~~~~~~~~~~~~~~~~~~~~~~~
       > src/media.m:54:44: warning: instance method '-UTF8String' not found (return type defaults to 'id') [-Wobjc-method-access]
       >       const char* album_cstr = [self.album UTF8String];
       >                                            ^~~~~~~~~~
       > /nix/store/km68nqhsk0h20ajnlqys3cnlqshs7cgv-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:238:8: note: receiver is instance of class declared here
       > @class NSString;
       >        ^
       > src/media.m:54:19: warning: incompatible pointer types initializing 'const char *' with an expression of type 'id' [-Wincompatible-pointer-types]
       >       const char* album_cstr = [self.album UTF8String];
       >                   ^            ~~~~~~~~~~~~~~~~~~~~~~~
       > src/media.m:99:32: error: property 'userInfo' not found on object of type 'const id'
       >       self.app = [notification.userInfo valueForKey:kMRMediaRemoteNowPlayingApplicationDisplayNameUserInfoKey];
       >                                ^
       > src/media.m:100:27: warning: instance method '-valueForKey:' not found (return type defaults to 'id') [-Wobjc-method-access]
       >       self.artist = [dict valueForKey:kMRMediaRemoteNowPlayingInfoArtist];
       >                           ^~~~~~~~~~~
       > /nix/store/km68nqhsk0h20ajnlqys3cnlqshs7cgv-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:244:8: note: receiver is instance of class declared here
       > @class NSDictionary;
       >        ^
       > src/media.m:101:26: warning: instance method '-valueForKey:' not found (return type defaults to 'id') [-Wobjc-method-access]
       >       self.title = [dict valueForKey:kMRMediaRemoteNowPlayingInfoTitle];
       >                          ^~~~~~~~~~~
       > /nix/store/km68nqhsk0h20ajnlqys3cnlqshs7cgv-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:244:8: note: receiver is instance of class declared here
       > @class NSDictionary;
       >        ^
       > src/media.m:102:26: warning: instance method '-valueForKey:' not found (return type defaults to 'id') [-Wobjc-method-access]
       >       self.album = [dict valueForKey:kMRMediaRemoteNowPlayingInfoAlbum];
       >                          ^~~~~~~~~~~
       > /nix/store/km68nqhsk0h20ajnlqys3cnlqshs7cgv-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:244:8: note: receiver is instance of class declared here
       > @class NSDictionary;
       >        ^
       > 11 warnings and 6 errors generated.
       > make: *** [makefile:61: bin/media.om] Error 1
       For full logs, run 'nix log /nix/store/4pas9210ydg0nw2mij5mclx90p73x412-sketchybar-2.16.0.drv'.

```

--- a/pkgs/os-specific/darwin/sketchybar/default.nix
+++ b/pkgs/os-specific/darwin/sketchybar/default.nix
@@ -1,10 +1,12 @@
 { lib
 , stdenv
 , fetchFromGitHub
-, Carbon
-, Cocoa
+, AppKit
+, CoreAudio
 , CoreWLAN
+, CoreVideo
 , DisplayServices
+, IOKit
 , MediaRemote
 , SkyLight
 }:
@@ -18,20 +20,22 @@ let
 in
 stdenv.mkDerivation (finalAttrs: {
   pname = "sketchybar";
-  version = "2.15.2";
+  version = "2.16.1";
 
   src = fetchFromGitHub {
     owner = "FelixKratz";
     repo = "SketchyBar";
     rev = "v${finalAttrs.version}";
-    hash = "sha256-13wc+1IgplB+L0j1AbBr/MUjEo4W38ZgJwrAhbdOroE=";
+    hash = "sha256-H+bR5ZhUTrN2KAEdY/hnq6c3TEb1NQvPQ9uPo09gSM8=";
   };
 
   buildInputs = [
-    Carbon
-    Cocoa
+    AppKit
+    CoreAudio
     CoreWLAN
+    CoreVideo
     DisplayServices
+    IOKit
     MediaRemote
     SkyLight
   ];

--- a/pkgs/os-specific/darwin/sketchybar/default.nix
+++ b/pkgs/os-specific/darwin/sketchybar/default.nix
@@ -57,6 +57,7 @@ stdenv.mkDerivation (finalAttrs: {
     description = "A highly customizable macOS status bar replacement";
     homepage = "https://github.com/FelixKratz/SketchyBar";
     license = lib.licenses.gpl3;
+    mainProgram = "sketchybar";
     maintainers = with lib.maintainers; [ azuwis khaneliman ];
     platforms = lib.platforms.darwin;
   };

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -13100,7 +13100,7 @@ with pkgs;
   sixpair = callPackage ../tools/misc/sixpair { };
 
   sketchybar = darwin.apple_sdk_11_0.callPackage ../os-specific/darwin/sketchybar {
-    inherit (darwin.apple_sdk_11_0.frameworks) Carbon Cocoa CoreWLAN DisplayServices MediaRemote SkyLight;
+    inherit (darwin.apple_sdk_11_0.frameworks) AppKit CoreAudio CoreWLAN CoreVideo DisplayServices IOKit MediaRemote SkyLight;
   };
 
   sketchybar-app-font = callPackage ../data/fonts/sketchybar-app-font { };

